Gitweb:     http://git.kernel.org/linus/59d069eb5ae9b033ed1c124c92e1532c4a958991
Commit:     59d069eb5ae9b033ed1c124c92e1532c4a958991
Parent:     b498ce1f2753b9724b2fc05d2057f7d1490cfa93
Author:     Xiao Guangrong <xiaoguangrong@cn.fujitsu.com>
AuthorDate: Tue Dec 1 17:30:08 2009 +0800
Committer:  Ingo Molnar <mingo@elte.hu>
CommitDate: Tue Dec 1 11:19:07 2009 +0100

    perf_event: Initialize data.period in perf_swevent_hrtimer()
    
    In current code in perf_swevent_hrtimer(), data.period is not
    initialized, The result is obvious wrong:
    
     # ./perf record -f -e cpu-clock make
     # ./perf report
     # Samples: 1740
     #
     # Overhead   Command                                   ......
     # ........  ........  ..........................................
     #
       1025422183050275328.00%        sh  libc-2.9.90.so ...
       1025422183050275328.00%      perl  libperl.so     ...
       1025422168240043264.00%      perl  [kernel]       ...
       1025422030011210752.00%      perl  [kernel]       ...
    
    Signed-off-by: Xiao Guangrong <xiaoguangrong@cn.fujitsu.com>
    Acked-by: Peter Zijlstra <peterz@infradead.org>
    Cc: Frederic Weisbecker <fweisbec@gmail.com>
    Cc: <stable@kernel.org>
    LKML-Reference: <4B14E220.2050107@cn.fujitsu.com>
    Signed-off-by: Ingo Molnar <mingo@elte.hu>
---
 kernel/perf_event.c |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/kernel/perf_event.c b/kernel/perf_event.c
index 0b9ca2d..040ee51 100644
--- a/kernel/perf_event.c
+++ b/kernel/perf_event.c
@@ -4010,6 +4010,7 @@ static enum hrtimer_restart perf_swevent_hrtimer(struct hrtimer *hrtimer)
 	event->pmu->read(event);
 
 	data.addr = 0;
+	data.period = event->hw.last_period;
 	regs = get_irq_regs();
 	/*
 	 * In case we exclude kernel IPs or are somehow not in interrupt
